src/codegen/compile.ml

```diff
@@ -6506,26 +6235,44 @@ module MakeSerialization (Strm : Stream) = struct
   *)
 
   module Strm = Strm
-
-  (* Globals recording known Candid types
-     See Note [Candid subtype checks]
-   *)
-
-  let register_delayed_globals env =
-    (E.add_global32_delayed env "__typtbl" Immutable,
-     E.add_global32_delayed env "__typtbl_end" Immutable,
-     E.add_global32_delayed env "__typtbl_size" Immutable,
-     E.add_global32_delayed env "__typtbl_idltyps" Immutable)
-
-  let get_typtbl env =
-    G.i (GlobalGet (nr (E.get_global env "__typtbl")))
-  let get_typtbl_size env =
-    G.i (GlobalGet (nr (E.get_global env "__typtbl_size")))
-  let get_typtbl_end env =
-    G.i (GlobalGet (nr (E.get_global env "__typtbl_end")))
-  let get_typtbl_idltyps env =
-    G.i (GlobalGet (nr (E.get_global env "__typtbl_idltyps")))
-
+    (* Globals recording known Candid types
+      See Note [Candid subtype checks]
+    *)
+    let register_delayed_globals env =
+      (E.add_global32_delayed env "__candid_data_length" Immutable,
+      E.add_global32_delayed env "__type_offsets_length" Immutable,
+      E.add_global32_delayed env "__idl_types_length" Immutable)
+
+    let get_candid_data_length env =
+      G.i (GlobalGet (nr (E.get_global env "__candid_data_length")))
+    let get_type_offsets_length env =
+      G.i (GlobalGet (nr (E.get_global env "__type_offsets_length")))
+    let get_idl_types_length env =
+      G.i (GlobalGet (nr (E.get_global env "__idl_types_length")))
+
+    let candid_type_offset_size = 4l
+
+    let get_global_type_descriptor env =
+      match !(E.(env.global_type_descriptor)) with
+      | Some descriptor -> descriptor
+      | None -> assert false
+
+    let load_candid_data env =
+      let descriptor = get_global_type_descriptor env in
+      Blob.load_data_segment env E.(descriptor.candid_data_segment) (get_candid_data_length env)
```

Comment:
Can you use Blob.constant here to avoid reallocation (e.g at each call to idl_sub)

Reply:
Thanks. I changed it use object pooling for the blob to avoid repeated Blob creation/loading. (I cannot use `Blob.constant` directly but need to apply `Tagged.share` since the data segment payload is only known at the module conclusion.)